freestyle-fs2/shared/src/main/scala/fs2.scala

```diff
@@ -0,0 +1,63 @@
+package freestyle
+
+import _root_.fs2._
+import _root_.fs2.util.{Attempt, Catchable, Free}
+
+import cats._
+import cats.free.{Free => CFree}
+import cats.arrow.FunctionK
+
+import scala.concurrent._
+
+object fs2 {
+  type Eff[A] = Free[CFree[Attempt, ?], A]
+
+  implicit val catsFreeC: Catchable[CFree[Attempt, ?]] = new Catchable[CFree[Attempt, ?]] {
+    def pure[A](a: A): CFree[Attempt, A]                              = CFree.pure(a)
+    def attempt[A](fa: CFree[Attempt, A]): CFree[Attempt, Attempt[A]] = fa.map(Attempt(_))
+    def fail[A](err: Throwable): CFree[Attempt, A]                    = CFree.liftF(Left(err))
+    def flatMap[A, B](a: CFree[Attempt, A])(f: A => CFree[Attempt, B]): CFree[Attempt, B] =
+      a.flatMap(f)
+  }
+
+  implicit val effC: Catchable[Eff] = new Catchable[Eff] {
+    def pure[A](a: A): Eff[A]                            = Free.pure(a)
+    def attempt[A](fa: Eff[A]): Eff[Attempt[A]]          = fa.attempt
+    def fail[A](err: Throwable): Eff[A]                  = Free.fail(err)
+    def flatMap[A, B](a: Eff[A])(f: A => Eff[B]): Eff[B] = a.flatMap(f)
+  }
+
+  @free sealed trait StreamM[F[_]] {
+    def run[A](s: Stream[Eff, A]): FreeS.Par[F, Unit]
+    def runLog[A](s: Stream[Eff, A]): FreeS.Par[F, Vector[A]]
+    def runFold[A, B](z: B, f: (B, A) => B)(s: Stream[Eff, A]): FreeS.Par[F, B]
+    def runLast[A](s: Stream[Eff, A]): FreeS.Par[F, Option[A]]
+  }
+
+  object implicits {
```

Comment:
The signature of `liftFSPar` can't be implemented with `Stream` since the only way to turn a `Stream[_[_], A]` into `A` is to fold over the stream.

```scala
    class Fs2FreeSLift[F[_]: StreamM] extends FreeSLift[F, Stream[Eff, ?]] {
      def liftFSPar[A](s: Stream[Eff, A]): FreeS.Par[F, A] = ???
    }
```